rasa/nlu/training_data/formats/rasa_yaml.py

```diff
@@ -0,0 +1,295 @@
+        from rasa.nlu.training_data import TrainingData
+
+        self._initialize()
+
+        yaml_content = io_utils.read_yaml(string)
+
+        for key, value in yaml_content.items():  # pytype: disable=attribute-error
+            if key == KEY_NLU:
+                self._parse_nlu(value)
+
+        return TrainingData(
+            self.training_examples,
+            self.entity_synonyms,
+            self.regex_features,
+            self.lookup_tables,
+        )
+
+    def _parse_nlu(self, nlu_data: List[Dict[Text, Any]]) -> None:
+
+        for nlu_item in nlu_data:
+            if not isinstance(nlu_item, dict):
+                raise_warning(
+                    f"Unexpected block found in '{self.filename}':\n"
+                    f"{nlu_item}\n"
+                    f"Items under the '{KEY_NLU}' key must be YAML dictionaries. "
+                    f"This block will be skipped. ",
+                    docs=DOCS_URL_TRAINING_DATA_NLU,
+                )
+                continue
+
+            for key in nlu_item.keys():
+                if key == KEY_INTENT:
+                    self._parse_intent(nlu_item)
+                elif key == KEY_SYNONYM:
+                    self._parse_synonym(nlu_item)
+                elif key == KEY_REGEX:
+                    self._parse_regex(nlu_item)
+                elif key == KEY_LOOKUP:
+                    self._parse_lookup(nlu_item)
+                else:
+                    raise_warning(
+                        f"Unexpected key '{key}' found in '{self.filename}'."
+                        f"Supported keys are: '{KEY_INTENT}', '{KEY_SYNONYM}', "
+                        f"'{KEY_REGEX}', '{KEY_LOOKUP}'. "
+                        f"This key will be skipped.",
+                        docs=DOCS_URL_TRAINING_DATA_NLU,
+                    )
+
+    def _parse_intent(self, data: Dict[Text, Any]) -> None:
+        from rasa.nlu.training_data import Message
+        import rasa.nlu.training_data.entities_parser as entities_parser
+        import rasa.nlu.training_data.synonyms_parser as synonyms_parser
+
+        intent = data.get(KEY_INTENT, "")
+        if not intent:
+            raise_warning(
+                f"Issue found while processing '{self.filename}':"
+                f"The intent has an empty name."
+                f"Intents should have a name defined under the {KEY_INTENT}"
+                "It will be skipped.",
+                docs=DOCS_URL_TRAINING_DATA_NLU,
+            )
+            return
+
+        examples = data.get(KEY_INTENT_EXAMPLES, "")
+
+        for example, entities in self._parse_training_examples(examples, intent):
+
+            synonyms_parser.add_synonyms_from_entities(
+                example, entities, self.entity_synonyms
+            )
+
+            plain_text = entities_parser.replace_entities(example)
+
+            message = Message.build(plain_text, intent)
+            if entities:
+                message.set("entities", entities)
```

Comment:
`"entities"` is defined in `nlu.constants`

Reply:
👍